content/library/api/charts/altair_chart.md

```diff
@@ -5,3 +5,143 @@ description: st.altair_chart displays a chart using the Altair library.
+
+#### Step 1: Create the base chart
+
+In this example, we create a time-series chart to track the evolution of stock prices. The chart is interactive and contains a multi-line tooltip. Click [here](https://altair-viz.github.io/gallery/multiline_tooltip.html) to learn more about multi-line tooltips in Altair.
+
+First, we import the required libraries and load the example stocks dataset using the [`vega_datasets`](https://pypi.org/project/vega-datasets/) package:
+
+```python
+import altair as alt
+import pandas as pd
+import streamlit as st
+from vega_datasets import data
+
+source = data.stocks()
+source = source[source.date.gt("2004-01-01")]
+```
+
+Next, we define a function `get_chart()` to create the interactive time-series chart of the stock prices with a multi-line tooltip. The x-axis represents the date, and the y-axis represents the stock price.
+
+We then invoke `get_chart()` that takes the stock prices dataframe as an input and returns a chart object. This is going to be our base chart on which we will overlay the annotations in [Step 2](/library/api-reference/charts/st.altair_chart#step-2-annotate-the-chart).
+
+```python
+# Define the base time-series chart.
+def get_chart(data):
+    hover = alt.selection_single(
+        fields=["date"],
+        nearest=True,
+        on="mouseover",
+        empty="none",
+    )
+
+    lines = (
+        alt.Chart(data, title="Evolution of stock prices")
+        .mark_line()
+        .encode(
+            x="date",
+            y="price",
+            color="symbol",
+        )
+    )
+
+    # Draw points on the line, and highlight based on selection
+    points = lines.transform_filter(hover).mark_circle(size=65)
+
+    # Draw a rule at the location of the selection
+    tooltips = (
+        alt.Chart(data)
+        .mark_rule()
+        .encode(
+            x="yearmonthdate(date)",
+            y="price",
+            opacity=alt.condition(hover, alt.value(0.3), alt.value(0)),
+            tooltip=[
+                alt.Tooltip("date", title="Date"),
+                alt.Tooltip("price", title="Price (USD)"),
+            ],
+        )
+        .add_selection(hover)
+    )
+    return (lines + points + tooltips).interactive()
+
+chart = get_chart(source)
+```
+
+#### Step 2: Annotate the chart
+
+Now that we have our base chart, we can annotate it with text and an emoji. Let's overlay the 💬 emoji on top of the time-series chart at specifc points of interest. We want users to hover over the 💬 emoji to see the associated annotation text.
+
+For simplicity, let's annotate four specific dates and set the height of the annotations at constant value of `10`.
+
+<Tip>
+
+You can vary the horizontal and vertical postions of the annotations by replacing the hard-coded values with the output of Streamlit widgets! Click [here](/library/api-reference/charts/st.altair_chart#interactive-example) to jump to the live example below.
```

Comment:
I've incorporated your suggestion and preserved the first line about replacing hard-coded values with widgets. I want to keep it in because IMO it's good to drive home the point that one can make the app more interactive by replacing fixed values with widgets.